src/audio/host-zephyr.c

```diff
@@ -403,9 +406,22 @@ static uint32_t host_get_copy_bytes_normal(struct host_data *hd, struct comp_dev
 	if (!(hd->ipc_host.feature_mask & BIT(IPC4_COPIER_FAST_MODE)))
 		dma_copy_bytes = MIN(hd->period_bytes, dma_copy_bytes);
 
-	if (!dma_copy_bytes)
-		comp_info(dev, "no bytes to copy, available samples: %d, free_samples: %d",
-			  avail_samples, free_samples);
+	if (!dma_copy_bytes) {
+		static uint64_t nobytes_last_logged;
+		static unsigned int n_skipped;
+		uint64_t now = k_uptime_get();
+		uint64_t delta = now - nobytes_last_logged;
+
+		if (delta > SOF_MIN_NO_BYTES_INTERVAL_MS) {
+			nobytes_last_logged = now;
+			comp_info(dev,
+				  "no bytes to copy, %u such events in last %llu ms, available samples: %d, free_samples: %d",
+				  n_skipped, delta, avail_samples, free_samples);
```

Comment:
nitpick: in this context the `avail_samples` and `free_samples` are misleading. They are valid for this specific print, but they might have been different fort he skipped ones...

Reply:
@ujfalusi I can split this print into 2: keep first unchanged for this specific event, and `if (n_skipped)` an additional print with the number and time interval